.github/workflows/ecr-shared-deploy-dev.yml

```diff
@@ -0,0 +1,117 @@
+# Dev ECS Container Deployment for Fargate & Lambda
+name: Dev Container Deploy
+
+on:
+  workflow_call:
+    inputs:
+      AWS_REGION:
+        required: true
+        type: string
+      GHA_ROLE:
+        required: true
+        type: string
+      ECR:
+        required: true
+        type: string
+      FUNCTION:
+        required: false
+        type: string
+      PREBUILD:
+        required: false
+        type: string
+
+# Set defaults
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  deploy:
+    name: Build and Deploy to ECR
+    runs-on: ubuntu-latest
+    # This line adds a check for the user which is requesting the PR. As long as its not dependabot, we go ahead and run it. 
+    if: ${{ github.triggering_actor != 'dependabot[bot]' }}
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Check for Python or Ruby
+        # Check for the existence of files that set either the Python version or the Ruby version
+        # and capture this informaiton for later use in the job. These values are used for
+        # conditional steps below to setup/configure the build environment for the container.
+        run: |
+          if [[ -f ".python-version" ]]; then
+            echo "python_repo=1" >> $GITHUB_ENV
+          else [[ -f ".ruby-version" ]]; 
```

Comment:
It does look a little weird, but in my head I was thinking "it's either a Python repo or a Ruby repo or a simple Docker repo", so the if/else instead of if/elif felt okay. So far, in my testing, it has worked as expected (the [docker-matomo](https://github.com/MITLibraries/docker-matomo) repo which has neither a .`ruby-version` nor a `.python-version` file doesn't attempt to prep either Python or Ruby).

That said, I'm totally comfortable switching this to 

```bash
if [[ -f ".python-version" ]]; then
    echo "python_repo=1" >> $GITHUB_ENV
fi
if [[ -f ".ruby-version" ]]; then
    echo "ruby_repo=1" >> $GITHUB_ENV
fi
```